spyder_terminal/widgets/main_widget.py

```diff
@@ -0,0 +1,451 @@
+        self.project_path = None
+        self.current_file_path = None
+        self.current_cwd = os.getcwd()
+
+        # Widgets
+        self.main = parent
+        self.find_widget = FindTerminal(self)
+        self.find_widget.hide()
+
+        layout = QVBoxLayout()
+
+        # Tab Widget
+        self.tabwidget = Tabs(self, rename_tabs=True)
+        self.tabwidget.currentChanged.connect(self.refresh_plugin)
+        self.tabwidget.move_data.connect(self.move_tab)
+        self.tabwidget.set_close_function(self.close_term)
+
+        if (hasattr(self.tabwidget, 'setDocumentMode') and
+                not sys.platform == 'darwin'):
+            # Don't set document mode to true on OSX because it generates
+            # a crash when the console is detached from the main window
+            # Fixes Issue 561
+            self.tabwidget.setDocumentMode(True)
+        layout.addWidget(self.tabwidget)
+        layout.addWidget(self.find_widget)
+        self.setLayout(layout)
+
+        self.__wait_server_to_start()
+
+    # ---- PluginMainWidget API
+    # ------------------------------------------------------------------------
+    def get_focus_widget(self):
+        """
+        Set focus on current selected terminal.
+
+        Return the widget to give focus to when
+        this plugin's dockwidget is raised on top-level.
+        """
+        term = self.tabwidget.currentWidget()
+        if term is not None:
+            return term.view
+
+    def get_title(self):
+        """Define the title of the widget."""
+        return _('Terminal')
+
+    def setup(self):
+        """Perform the setup of plugin's main menu and signals."""
+        self.cmd = find_program(self.get_conf('shell'))
+        self.server = subprocess.Popen(
+            [sys.executable, '-m', 'spyder_terminal.server',
+             '--port', str(self.port), '--shell', self.cmd],
+            stdout=self.server_stdout,
+            stderr=self.server_stderr)
+        self.color_scheme = self.get_conf('appearance', 'ui_theme')
+        self.theme = self.get_conf('appearance', 'selected')
+
+        # Menu
+        menu = self.get_options_menu()
+
+        # Actions
+        new_terminal_action = self.create_toolbutton(
+            TerminalContainerWidgetToolbarSections.New,
+            text=_("Open a new terminal"),
+            icon=self.create_icon('expand_selection'),
+            triggered=lambda: self.create_new_term(),
+        )
+
+        self.add_corner_widget(
+            TerminalContainerWidgetCornerToolbar.NewTerm,  new_terminal_action)
+
+        new_terminal_cwd = self.create_action(
+            TerminalContainerWidgetActions.NewTerminalCWD,
+            text=_("New terminal in current working directory"),
+            tip=_("Sets the pwd at the current working directory"),
+            triggered=lambda: self.create_new_term(),
+            shortcut_context='terminal',
+            register_shortcut=True)
```

Comment:
I think we should register this shortcut for `new_terminal_action`. Is that not possible?